tests/project_template/test_settings.py

```diff
@@ -1,30 +1,33 @@
 import os
 import shutil
+import tempfile
 
 from django import conf
 from django.test import TestCase
+from django.test.utils import extend_sys_path
 
 
 class TestStartProjectSettings(TestCase):
     def setUp(self):
-        # Ensure settings.py exists
-        project_dir = os.path.join(
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.addCleanup(self.temp_dir.cleanup)
```

Comment:
Oh! I didn't see the *initial* question of:

> I wonder if this is needed since the TemporaryDirectory is deleted when garbage collected?

Now — with this context! —  the finalizer comments make a little more sense. (Yes, I deliberately cleared it up as I understood that to be the case re. weakrefs, etc etc.)